### ISSUE TITLE
Passenger trying to create `NGX_HTTP_PROXY_TEMP_PATH` folder.

### ISSUE THREAD

**Issue by CamJN** [phusion/passenger#2075]
The module creates a folder which is not used by Passenger, even when nginx configuration overrides `NGX_HTTP_PROXY_TEMP_PATH`.

configuration slice:
```
http {
client_body_temp_path /var/nginx/client_body_temp;
proxy_temp_path /var/nginx/proxy_temp;
fastcgi_temp_path /var/nginx/fastcgi_temp;
uwsgi_temp_path /var/nginx/uwsgi_temp;
scgi_temp_path /var/nginx/scgi_temp;
}
```
strace log:
```
open("/var/run/nginx/nginx.pid", O_RDWR|O_CREAT, 0644) = 4
close(4) = 0
mkdir("/var/nginx/client_body_temp", 0700) = -1 EEXIST (File exists)
mkdir("/var/nginx/proxy_temp", 0700) = -1 EEXIST (File exists)
mkdir("/var/nginx/fastcgi_temp", 0700) = -1 EEXIST (File exists)
mkdir("/var/nginx/uwsgi_temp", 0700) = -1 EEXIST (File exists)
mkdir("/var/nginx/scgi_temp", 0700) = -1 EEXIST (File exists)
mkdir("/opt/nginx/proxy_temp", 0700) = -1 EACCES (Permission denied)
write(3, "2018/05/08 10:17:55 [emerg] 3740"..., 100) = 100
write(2, "nginx: [emerg] mkdir() \"/opt/nginx/proxy_temp\" failed (13: Permission denied)"..., 78) = 78
```
probably caused by `src/nginx_module/Configuration.c` line 64:
```
static ngx_path_init_t ngx_http_proxy_temp_path = {
ngx_string(NGX_HTTP_PROXY_TEMP_PATH), { 1, 2, 0 }
};
```

**Comment by ROMB** [comment]
I actually submitted an easy PR #2074 to fix this.

**Comment by CamJN** [comment]
Thanks for that, we'll review this as soon as we can. In the meantime could you sign the [contributor agreement](https://www.phusion.nl/contributor).

**Comment by ROMB** [comment]
Already signed

**Comment by FooBarWidget** [comment]
Thanks for submitting the pull request @ROMB. I'm wondering what other effects there are of removing that code. What will happen to the buffering behavior on the Nginx side?

**Comment by ROMB** [comment]
I don't really know, not that much expert in nginx. If this variable and folder actually needed for buffering to work, than someone need to make that folder configurable setting, which I will not be able to do because of lack of experience.

**Comment by CamJN** [comment]
@ROMB are you comfortable building passenger from source to test out a config option? It seems to work for me, but I'd like more eyes on it.

**Comment by halo** [comment]
I just wanted to ask whether there is any work-around for this problem?

**Background:**

After upgrading, I cannot start nginx (1.17.6) with passenger (6.0.4) because it tries to create this subdirectory, but I don't have a `/usr/local/var` directory (and I don't need one, because nginx is installed at a custom location using homebrew)

```bash
nginx: [emerg] mkdir() "/usr/local/var/run/nginx/proxy_temp" failed (2: No such file or directory)
```

Setting `proxy_buffering off;` in my nginx.conf had no effect.

**Comment by ROMB** [comment]
I've build passenger from sources using patch provided in mentioned PR,
unfortunately that's the only option for now.
@CamJN can provide you with better patch to test, which makes this folder
configurable.

**Comment by CamJN** [comment]
Yes, the HEAD of the repo slready has the patch merged, or you can apply the patch at https://github.com/phusion/passenger/commit/e512231f.patch?full_index=1 to 6.0.4. Then the option to configure is called `passenger_temp_path`.
